### DISCUSSION TITLE
cash back

### DISCUSSION THREAD

**Discussion by scoobiii** [bacen/pix-api#255] [Ideas]
Olá,
API PLIM Monetizador da economia circular no ato da compra

Dúvida: Como programar e aplicar cashback no pix?

Atores:

Cliente:
Vendedor:  de marreteiro vendendo refrigerante cerveja agua na rua a Hipermercado Físico ou Marktplace passando por qualquer coisa produzida e embalada no pib global.
Agente de Reciclagem Móvel: o homem do saco, carroceiro, tiozinho da pirua, ..., ..., ...., ......
Agente de Reciclagem Fixo: caçambas esparramadas pela cidade,..,supermercados,..,.ecoponto$..,....S.A

Transações: R - CVREP - Pensa em Reciclar, depois.... Comprar, Vender, Receber, Entregar e Pagar
 
C (familia B ou C, A, ...) compra R$1600,00 mensalmente de V
V vende pix de R$1600,00 sendo R$16,00 (1% valor intrínseco das embalagens. Ref. O mercado paga R$0,05 por latinha vazia de coca cola de alumínio, ou 1% do valor do produto)
C recebe produtos de V
C paga pix de R$1600 a V
C consome os produtos, entrega as embalagens a ARF.
ARF paga R$16 pix a C
C, com preguiça, aciona ARM
C paga R$16 + custo logístico de deslocamento do ARM. Sim, enquanto o homem do saco (garimpadores) e carroceiros não forem remunerado pelo deslocamento, assim como o Uber é, continuarão invisíveis e fora do sistema financeiro.
ARM recebe R$16 pix e a distancia logística

https://github.com/scoobiii/hackathon-apicon-2020-plimm
https://github.com/scoobiii/PLIMM-Bank

**Comment by rubenskuhl** [comment]
Me parece que o split payment já comentado em #66 e #229 atenderia o cashback imediato, e que os posteriores poderiam ser atendidos pelo #254 . 
O resto me parecem regras de negócios de uma cadeia de valor específica.

**Reply by scoobiii** [reply]
@rubenskuhl avançamos. obrigado.

// meta. menos de 20 linhas.

//transação 1

{

 "produto_cód": "5100446",
 "produto_descrição":"Cerveja HEINEKEN Garrafa 600 ml",
  "valor": "9,90", // ou R$1600,00 compras supermercado familia classe B
  "cashbak_plimm_embalagem":"0.099",  // ou R$16,00
  "pagador_plimm_cliente": {
    "chave": "plimm_cliente@sua_conta.com.br"
  },
  "favorecido_plimm_vendedor_ambulante": {
    "chave": "plimm_vendedor_ambulante@sua_conta.com.br"
  }
}

OU

{
  "valor": "9,90",  // garrafa de cerveja, compras R$1600,00 supermercado ou avião 
  "pagador_plimm_cliente": {
    "chave": "plimm_cliente@sua_conta.com.br",
    "infoPagador": "Segue o pagamento da conta. Cashback/descontos R$0,99.  // ou R16,00 compras supermercado
    "infoPlimm": "Georeferência de pontos de coleta fixos e móveis. Deslocamento de catadores é descontado"
    
  },
  "favorecido_plimm_vendedor_ambulante": {                 // supermercado ou amazon
    "plimm_vendedor_ambulante@sua_conta.com.br": {
      "nome": "Nome Favorecido",
      "cpf": "00000000000",
      "codigoBanco": "09089356", 
      "agencia": "1",
      "conta": "123453",
      "tipoConta": "cacc"
    }
  }
}

--------------------------------------------------

transação 2

{
  "valor": "0,99",
  "pagador_plimm_vendedor_ambulante": {
    "chave": "plimm_vendedor_ambulante@sua_conta.com.br"
  },
  "favorecido_plimm_cliente": {
    "chave": "plimm_cliente@sua_conta.com.br"
  }
}

OU

usar um identificador de objeto na unha para assegurar que a latinha saiu daqui: https://github.com/scoobiii/mundrundo5.0

{
  "valor": "0,99",
  "pagador_plimm_vendedor_ambulante": {
    "chave": "plimm_vendedor_ambulante@sua_conta.com.br",
    "infoPagador_plimm_vendedor_ambulante": "Segue o pagamento de  0,99 PLIMM´s. Parabéns por reciclar no ato da compra!!"
  },
  "favorecido_plimm_cliente": {
    "plimm_cliente@sua_conta.com.br": {


   //   0,01% por operação, de cash back lastreado em embalagens. ref. lata coca cola R$,05 pago pelo mercado/indústria.

      "nome": "Nome Favorecido",
      "cpf": "00000000000",
      "codigoBanco": "09089356", 
      "agencia": "1",
      "conta": "123453",
      "tipoConta": "cacc"
    }
  }
}

**Reply by rubenskuhl** [reply]
Acho que há uma questão de timing que precisa estar clara: o que tem hoje disponível na API Pix é apenas cobrança. Idéias de split e de pagamento via API são... idéias. Talvez a que fique pronta antes é o pagamento citado na #254, pois ele tem utilidade para mais casos de uso. 

Não há serviço de cobrança via API Pix sendo oferecido a 1 centavo por cobrança(exceto se a cobrança for de R$1 ou menos). Talvez você consiga convencer algum PSP a cobrar isso desse projeto específico dado seu impacto social, mas os valores típicos são bem maiores que esse. Há até vários PSPs não cobrando nada nesses primeiros meses do Pix, então essa janela pode ser usada para colocar algo no ar e depois achar uma "casa hospitaleira". 

Você pode ver quais PSPs tem API Pix numa lista que tento manter com auxílio da comunidade em #76.

**Reply by scoobiii** [reply]
aqui o psp eh o ambulante operando na unha mensagens. replicado numa rede de supermercados, onde a barreira eh o crm impenetravel, onde rest pode trazer o que eles não querem soltar, resolvido com grapql na api que libera só o que eles querem, e o que nós precismos, as embalagens, então a escala justifica.
social. carroceiro e uber são sub empregos, mas o segundo conta com api remunerando seu deslocamento de a to b (vazio) dividindo a conta global entre este e a conta de luz do servidor, então é timing cultural enchergar e remunerar 25% para o server e75% de R$7,60, o deslocamento do catador/carroceiro mais os R$16 de cashback lastreado em compras, ou seja, juntar dois meses para equilibra.
.sem remunerar o carroceiro pelo deslocamento, ele continua invisivel....e...nécatibiriba bancarização.

não tem almoço de graça, então, o pix vai cobrar, e se for R$0,01, de cada pf, vai ta de bom tamanho. a internet começou assim. de graça. conhecimento nas universidades/publicas, agora fechado o acesso....cobrando alguns cents....briga e destino do menino da internet...

massa essa lista!!!

**Reply by rubenskuhl** [reply]
PSP é um jargão do Pix, Prestador de Serviço de Pagamentos. É quem tem a conta transacional, tanto do recebedor quanto do pagador. Você vai precisar de um para fazer o recebimento e cada um para quem você quer repassar vai também precisar ter conta transacional. Notar que conta transacional inclui conta corrente, conta poupança e conta pagamento, mais conhecida na praça como carteira digital. E enquanto conta corrente há menos casos de gratuidade, conta poupança e conta pagamento tem muitos exemplos de contas gratuitas que tem sido usadas para incluir os desbancarizados. Talvez o mais difícil seja ter um celular para abrir e movimentar a conta... ou ter comprovante de endereço, que muitos desses processos pedem.

**Reply by scoobiii** [reply]
olhando a lista #76 com carinho.
primeiro mainframe foi do itaú. 
clientes: zero mips
10 mips x 18 MM clientes --->  ctmm mogimirim----> vendido/terceirizado ---> cliente datacenter 'apple pra sempre' = $18k

<--- agente de reciclagem (carroceiro)  bancarizado compra a vista. primeiro remunerar seu deslocamento: ....cartão, relógio, iot siri (mini arduino, esp8266) + geolocation...
cripto pix II?

**Reply by scoobiii** [reply]
https://opensea.io/collection/backplimm criptoplimm devolvidos quando as embalagens percorrem a rota da economia circular


https://opensea.io/collection/bottlewall destinação lúdica para embalagens.


APP PLIMM | API de pagamento | Cartão e Conta Digital PLIMM | Geolocation: Cadastra e integra meios de pagamentos múltiplos, cadeia de distribuição, indústria e agentes de reciclagem.

**Comment by scoobiii** [comment]
tem o perfil de time ninja pra codar, uxui, gp agile, tempo, $,mkt,operwcoes, ceo, roi, ipo by pink sheet 
onde vc entra?

**Comment by scoobiii** [comment]
salve seus lindos,

por falar capim, a estrutura básica, segue o ecossistema verticalizado plimm, com moeda plimm, base do cashbackplimm, que conversa com todas moedas..
__Não há serviço de cobrança via API Pix sendo oferecido a 1 centavo por cobrança(exceto se a cobrança for de R$1 ou menos). Talvez você consiga convencer algum PSP a cobrar isso desse projeto específico dado seu impacto social, mas os valores típicos são bem maiores que esse. Há até vários PSPs não cobrando nada nesses primeiros meses do Pix, então essa janela pode ser usada para colocar algo no ar e depois achar uma "casa hospitaleira".__

o pix ja aceita transações em centavos para rastreamento conversibilidade automática entre 

plimm == eth == pix analógico == pix digital == pix chain

como anda a implementação do pixchain?

abraços
feliz ano novo \o/
2024, carpim per capita de 30 ETH para gado de esquerda centro e direita.
paz no campo.

--------------------------------------------------------------------------------------------------
#dezembro 2023, classe ecossistema plimm versão 1.0.5, zeh sobrinho, @poplixo

-record(banco_plimm, {nome, cripto_conta, posicao_geografica, conta_energia}).
-record(cartao_credito_carbono_debito, {numero, titular, validade, cripto_conta}).
-record(cliente_plimm, {nome, cripto_conta, posicao_geografica, conta_energia}).
-record(residuo_tecnico_plimm, {peso, consumo_energia_tonelada, cripto_conta}).
-record(garrafa_plimm, {marca, modelo, rotulo, ano, produtor, origem, qr_code, casco_plimm, peso, preco, consumo_energia_garrafa, cripto_conta}).
-record(supermercado_plimm, {nome, cripto_conta, posicao_geografica, conta_energia}).
-record(catador_plimm, {nome, cripto_conta, posicao_geografica, conta_energia}).
-record(cooperativa_catadores_plimm, {nome, cripto_conta, posicao_geografica, conta_energia}).
-record(recicladora_plimm, {nome, cripto_conta, posicao_geografica, conta_energia, consumo_energia_garrafa}).
-record(fabrica_garrafa_plimm, {nome, cripto_conta, posicao_geografica, conta_energia, consumo_energia_tonelada}).
-record(engarrafador_plimm, {nome, cripto_conta, posicao_geografica, conta_energia, consumo_energia_garrafa}).
-record(produtor_energia_plimm, {nome, cripto_conta, posicao_geografica, conta_energia, tipo_energia, eficiencia}).
-record(hidraulica_plimm, {eficiencia}).
-record(solar_plimm, {eficiencia}).
-record(eolica_plimm, {nome, cripto_conta, posicao_geografica, conta_energia, eficiencia, altura_torre}).
-record(termica_plimm, {nome, tipo_combustivel, proporcao_combustivel, eficiencia}).
-record(hibrida_plimm, {nome, tipo_combustivel1, proporcao_combustivel1, proporcao_combustivel2, eficiencia}).

%% banco_plimm
-record(banco_plimm, {nome, cripto_conta, posicao_geografica, conta_energia}).

new_banco_plimm(Nome, CriptoConta, PosicaoGeografica, ContaEnergia) ->
    #banco_plimm{nome = Nome, cripto_conta = CriptoConta, posicao_geografica = PosicaoGeografica, conta_energia = ContaEnergia}.

%% cartao_credito_carbono_debito
-record(cartao_credito_carbono_debito, {numero, titular, validade, cripto_conta}).

new_cartao_credito_carbono_debito(Numero, Titular, Validade, CriptoConta) ->
    #cartao_credito_carbono_debito{numero = Numero, titular = Titular, validade = Validade, cripto_conta = CriptoConta}.

%% cliente_plimm
-record(cliente_plimm, {nome, cripto_conta, posicao_geografica, conta_energia}).

new_cliente_plimm(Nome, CriptoConta, PosicaoGeografica, ContaEnergia) ->
    #cliente_plimm{nome = Nome, cripto_conta = CriptoConta, posicao_geografica = PosicaoGeografica, conta_energia = ContaEnergia}.

%% residuo_tecnico_plimm
-record(residuo_tecnico_plimm, {peso, consumo_energia_tonelada, cripto_conta}).

new_residuo_tecnico_plimm(Peso, ConsumoEnergiaTonelada, CriptoConta) ->
    #residuo_tecnico_plimm{peso = Peso, consumo_energia_tonelada = ConsumoEnergiaTonelada, cripto_conta = CriptoConta}.

%% garrafa_plimm
-record(garrafa_plimm, {marca, modelo, rotulo, ano, produtor, origem, qr_code, casco_plimm, peso, preco, consumo_energia_garrafa, cripto_conta}).

new_garrafa_plimm(Marca, Modelo, Rotulo, Ano, Produtor, Origem, QrCode, CascoPlimm, Peso, Preco, ConsumoEnergiaGarrafa, CriptoConta) ->
    #garrafa_plimm{marca = Marca, modelo = Modelo, rotulo = Rotulo, ano = Ano, produtor = Produtor, origem = Origem, qr_code = QrCode, casco_plimm = CascoPlimm, peso = Peso, preco = Preco, consumo_energia_garrafa = ConsumoEnergiaGarrafa, cripto_conta = CriptoConta}.

%% supermercado_plimm
-record(supermercado_plimm, {nome, cripto_conta, posicao_geografica, conta_energia}).

new_supermercado_plimm(Nome, CriptoConta, PosicaoGeografica, ContaEnergia) ->
    #supermercado_plimm{nome = Nome, cripto_conta = CriptoConta, posicao_geografica = PosicaoGeografica, conta_energia = ContaEnergia}.

%% catador_plimm
-record(catador_plimm, {nome, cripto_conta, posicao_geografica, conta_energia}).

new_catador_plimm(Nome, CriptoConta, PosicaoGeografica, ContaEnergia) ->
    #catador_plimm{nome = Nome, cripto_conta = CriptoConta, posicao_geografica = PosicaoGeografica, conta_energia = ContaEnergia}.

%% cooperativa_catadores_plimm
-record(cooperativa_catadores_plimm, {nome, cripto_conta, posicao_geografica, conta_energia}).

new_cooperativa_catadores_plimm(Nome, CriptoConta, PosicaoGeografica, ContaEnergia) ->
    #cooperativa_catadores_plimm{nome = Nome, cripto_conta = CriptoConta, posicao_geografica = PosicaoGeografica, conta_energia = ContaEnergia}.

%% recicladora_plimm
-record(recicladora_plimm, {nome, cripto_conta, posicao_geografica, conta_energia, consumo_energia_garrafa}).

new_recicladora_plimm(Nome, CriptoConta, PosicaoGeografica, ContaEnergia, ConsumoEnergiaGarrafa) ->
    #recicladora_plimm{nome = Nome, cripto_conta = CriptoConta, posicao_geografica = PosicaoGeografica, conta_energia = ContaEnergia, consumo_energia_garrafa = ConsumoEnergiaGarrafa}.

%% fabrica_garrafa_plimm
-record(fabrica_garrafa_plimm, {nome, cripto_conta, posicao_geografica, conta_energia, consumo_energia_tonelada}).

new_fabrica_garrafa_plimm(Nome, CriptoConta, PosicaoGeografica, ContaEnergia, ConsumoEnergiaTonelada) ->
    #fabrica_garrafa_plimm{nome = Nome, cripto_conta = CriptoConta, posicao_geografica = PosicaoGeografica, conta_energia = ContaEnergia, consumo_energia_tonelada = ConsumoEnergiaTonelada}.

%% engarrafador_plimm
-record(engarrafador_plimm, {nome, cripto_conta, posicao_geografica, conta_energia, consumo_energia_garrafa}).

new_engarrafador_plimm(Nome, CriptoConta, PosicaoGeografica, ContaEnergia, ConsumoEnergiaGarrafa) ->
    #engarrafador_plimm{nome = Nome, cripto_conta = CriptoConta, posicao_geografica = PosicaoGeografica, conta_energia = ContaEnergia, consumo_energia_garrafa = ConsumoEnergiaGarrafa}.

%% produtor_energia_plimm
-record(produtor_energia_plimm, {nome, cripto_conta, posicao_geografica, conta_energia, tipo_energia, eficiencia}).

new_produtor_energia_plimm(Nome, CriptoConta, PosicaoGeografica, ContaEnergia, TipoEnergia, Eficiencia) ->
    #produtor_energia_plimm{nome = Nome, cripto_conta = CriptoConta, posicao_geografica = PosicaoGeografica, conta_energia = ContaEnergia, tipo_energia = TipoEnergia, eficiencia = Eficiencia}.

%% hidraulica_plimm
-record(hidraulica_plimm, {eficiencia}).

new_hidraulica_plimm(Eficiencia) ->
    #hidraulica_plimm{eficiencia = Eficiencia}.

%% solar_plimm
-record(solar_plimm, {eficiencia}).

new_solar_plimm(Eficiencia) ->
    #solar_plimm{eficiencia = Eficiencia}.

%% eolica_plimm
-record(eolica_plimm, {nome, cripto_conta, posicao_geografica, conta_energia, eficiencia, altura_torre}).

new_eolica_plimm(Nome, CriptoConta, PosicaoGeografica, ContaEnergia, Eficiencia, AlturaTorre) ->
    #eolica_plimm{nome = Nome, cripto_conta = CriptoConta, posicao_geografica = PosicaoGeografica, conta_energia = ContaEnergia, eficiencia = Eficiencia, altura_torre = AlturaTorre}.

%% termica_plimm
-record(termica_plimm, {nome, tipo_combustivel, proporcao_combustivel, eficiencia}).

new_termica_plimm(Nome, CriptoConta, PosicaoGeografica, ContaEnergia, Eficiencia) ->
    #termica_plimm{nome = Nome, tipo_combustivel = CriptoConta, proporcao_combustivel = PosicaoGeografica, eficiencia = Eficiencia}.

%% hibrida_plimm
-record(hibrida_plimm, {nome, tipo_combustivel1, proporcao_combustivel1, proporcao_combustivel2, eficiencia}).
    
new_hibrida_plimm(Nome, CriptoConta, PosicaoGeografica, ContaEnergia, Eficiencia) ->
    #hibrida_plimm{nome = Nome, tipo_combustivel1 = CriptoConta, proporcao_combustivel1 = PosicaoGeografica, proporcao_combustivel2 = ContaEnergia, eficiencia = Eficiencia}.
